pop_single.m

```diff
@@ -706,52 +709,61 @@ function replace_parents(pop)
         % evaluate the objective function(s) correctly
         function evaluate_function(pop)
                    
-            % NOTE: suited for both single and multi-objective optimization
+            % multi-objective optimization overloads this function for
+            % some initialization, but returns here later.
             
             % find evaluation sites
             if isempty(pop.pop_data.function_values_offspring)
                 sites = 1:pop.size; % only in pop-initialization
+                fvs = zeros(length(sites), 1);  % Single-objective only
             else
                 % TODO: check if this works correctly; set as logicals
                 % here, but used as index below
                 sites = ~isfinite(pop.pop_data.function_values_offspring(:, 1));
+                fvs = zeros(nnz(sites), ...
+                            size(pop.pop_data.function_values_offspring, 2));
             end
             num_pop = nnz(sites);
-            
+
+            pop_inputs = ...
+              pop.pop_data.offspring_population(sites, :);
+
             % evaluate all functions for each population member in
             % parallel or in serial
-            fvs = pop.pop_data.offspring_population(sites, :);
             if pop.options.UseParallel
               parfor pop_num=1:num_pop
-                evaluate_one_function(pop, sites, pop_num);

```

Comment:
I have a feeling this way of parallelizing is inefficient...wouldn't it be better to evaluate the different algorithms in parallel, in their entirety? e.g., DE on 1 core, ASA on another, GA on the 3rd, etc. Feels like there should be less overhead that way, but profiling is required